pkg/router/template/router.go

```diff
@@ -990,6 +1011,35 @@ func (r *templateRouter) createServiceAliasConfig(route *routev1.Route, backendK
 	return &config
 }
 
+func getHeadersList(httpHeaderList []routev1.RouteHTTPHeader) []HTTPHeader {
+	var httpHeadersList []HTTPHeader
```

Comment:
How do you ensure that these headers don't override the headers that the cluster admin set in the ingress controller?  What if the cluster admin doesn't want users to set request/response headers to something different than what the admin set?

Reply:
The ingress controller updates the front end sections only and the route updates the plain backend section in the haproxy template. Both have different use cases.